python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -654,22 +657,32 @@ def run(self, *, exposures, id_generator=None, result=None):
                                                                                   id_generator)
             have_fit_psf = True
             self._measure_aperture_correction(result.exposure, result.psf_stars_footprints)
+            # Run astrometry using PSF candidate stars
+            astrometry_matches, astrometry_meta = self._fit_astrometry(
+                result.exposure, result.psf_stars_footprints
+            )
             result.psf_stars = result.psf_stars_footprints.asAstropy()
+            self.metadata["astrometry_matches_count"] = len(astrometry_matches)
+            if "astrometry_matches" in self.config.optional_outputs:
+                result.astrometry_matches = lsst.meas.astrom.denormalizeMatches(astrometry_matches,
+                                                                                astrometry_meta)
 
+            # Run the stars_detection subtask for the photometric calibration.
             result.stars_footprints = self._find_stars(result.exposure, result.background, id_generator)
             self._match_psf_stars(result.psf_stars_footprints, result.stars_footprints)
+
+            # Update the source cooordinates with the current wcs.
+            afwTable.updateSourceCoords(result.exposure.wcs, sourceList=result.stars_footprints)
```

Comment:
Consider whether you should also set `include_covariance=True` to update the coordinate error fields.

Reply:
I think that's the default: https://github.com/lsst/afw/blob/main/include/lsst/afw/table/wcsUtils.h#L64